### ISSUE TITLE
Replace console.logging with a mixture of loggable events and use of debug.

### ISSUE THREAD

**Issue by timexcession** [matrix-org/matrix-js-sdk#1970]
This SDK is a library intended for integration into third party applications.

However it spits out all sorts of internal debug information via `console.log` statements direct to stdout, via `logger.ts`

For our purposes, this is an anti pattern - we use `stdout` as the only 'appender' to which our managed services log.

It would be much healthier to make use of the debug library here - https://www.npmjs.com/package/debug for debug level statements.

Anything else that could be logged could instead be raised via an EventEmitter of some kind, and then application code can choose what to do with those events - log or discard - and also determine the application specific log level.

Alternatively - is it easy to turn off all logging statements within the library? This would be an acceptable, but less than ideal work around.

**Comment by bradjones1** [comment]
Agreed. I am also seeing quite a bit of `value.hasOwnProperty is not a function` when logging from React Native, which I think is due to the shape of some objects that are logged to the console from the Matrix library.

**Comment by bradjones1** [comment]
Related: https://github.com/matrix-org/matrix-js-sdk/pull/1611

**Comment by bradjones1** [comment]
Should actually follow the pattern from https://github.com/matrix-org/matrix-js-sdk/issues/332 ?
